### ISSUE TITLE
[BUG] No masks found after training

### ISSUE THREAD

**Issue by jzeng22** [MouseLand/cellpose#947]
Hi, 

I've been trying to do human-in-the-loop model training on the GUI. When I run the untrained cyto3 model on one of my images (changing the diameter to 10 pixels and the upper normalization percentage to 99.99 to account for the sparsity in my image), it works fine, and masks are found. Here is an example of the output of the model ([example_mask.zip](https://github.com/MouseLand/cellpose/files/15395443/example_mask.zip)). However, after training (by going to "Models" and then clicking on "Train new model with image+masks in folder" and using the default training parameters), no masks are found. I've tried training on multiple (4) images (running the untrained cyto3 model on multiple images and then training), but the issue remains. 

Attached are my images ([images.zip](https://github.com/MouseLand/cellpose/files/15395456/images.zip)) and the terminal output ([terminal_output.txt](https://github.com/MouseLand/cellpose/files/15395459/terminal_output.txt)). 

Thank you!

**Comment by lsxucsf** [comment]
Any resolution to this issue? I'm having the same problem- with cellpose 3.0.7. Rolling back to cellpose 2 returned the human-in-the-loop functionality as before.

**Comment by Math7914** [comment]
Same issue for me. No mask found after human-in-the-loop training. Waiting for the resolution

**Comment by carsen-stringer** [comment]
sorry I think I found the bug here, the cellpose3 GUI was only using 8
images per epoch during training with human-in-the-loop, can you please
install the latest version and try it out? `pip install git+
https://github.com/mouseland/cellpose.git`

**Comment by lsxucsf** [comment]
That fixed it for me. Thanks Carsen

**Comment by jzeng22** [comment]
I'm still getting 0 masks after training. Notably, when I use images downloaded from the cellpose website, the training process works fine (this was also true prior to installing the latest version).

**Comment by viditagr** [comment]
Hey @jzeng22, are you still having the same problem or was it resolved because I think I am also facing the same issues. I am getting an info message saying no cell pixels found which I believe is equivalent to saying no masks found. I am also trying to use the exact same workflow, also tried to update it as suggested but it did not fix it! Any help would be appreciated!

**Comment by aleximcadams** [comment]
Hi, I have also been having this issue where in Cellpose 3.0, either no masks are found when training or the quality is very very poor for most datasets I put through it with a ton of image/mask pairs. 

If I use the same exact folder with images and masks but train it in Cellpose 2.0, it works beautifully. I would just use Cellpose 2.0, except I also have the bug where I am unable to draw masks only in 2.0. It begins drawing normally but then does not finish if I get to the point where the mask started, and then that dot disappears and I can never complete drawing a mask. Both bugs for 3.0 and 2.0 are happening for other people I know. 

To get around this, I have been using Cellpose 2.0 to train, and then I edit the masks once it's been applied to the next image in the folder in Cellpose 3.0. Then I go back to 2.0 to train and repeat until the model is good enough. It works but would appreciate a better solution. Thank you!

**Comment by carsen-stringer** [comment]
sorry we had a bug where not all images were used for training in Cellpose3, very sorry about this! please upgrade `pip install git+https://github.com/mouseland/cellpose.git`

for the issue with drawing, please see issue #961 . this is a known issue but unfortunately I have not been able to replicate this bug myself on Windows, Linux and Mac, so I haven't been able to figure out the problem, I think if you have the latest python and the latest pyqt6 it should work? also use hard right-click on Mac.

if you're still having issues with training with the latest version I will reopen this issue

**Comment by DyangLi** [comment]
Hi I'm also encountering the same problem that no cell masks were identified after training. I tried updating Cellpose but is still isn't working. 
 
```
2024-09-10 14:53:50,743 [WARNING] WARNING: no mask pixels found
2024-09-10 14:53:50,743 [INFO] No cell pixels found.
2024-09-10 14:53:50,972 [INFO] 0 cells found with model in 1.282 sec
GUI_INFO: 0 masks found
GUI_INFO: creating cellcolors and drawing masks
```


My working environment is the following: 

```
2024-09-10 14:53:24,080 [INFO]
cellpose version: 	3.0.12.dev72+g385891c
platform:       	darwin
python version: 	3.10.13
torch version:  	2.3.0.dev20240216
2024-09-10 14:53:24,318 [INFO] ** TORCH MPS version installed and working. **
2024-09-10 14:53:35,929 [ERROR] <tifffile.TiffPages @8> invalid page offset 26395562
```

**Comment by carsen-stringer** [comment]
just to confirm @DyangLi this set of images worked with the Cellpose 2 GUI but not the Cellpose 3 GUI?

**Comment by DyangLi** [comment]
@carsen-stringer Yes can confirm it works without a problem with a 2.3.2 version. 
The problem seems to be in the training part. If I use v3-trained model it would fail in v2 as well. Also noticed that during training, v2 is giving non-zero loss term, but v3 is giving all zeros in both loss terms.

**Comment by carsen-stringer** [comment]
I think this could be an MPS related issue then -- can you try upgrading to the latest pytorch version? `pip install torch --upgrade`. I think 2.4 should support training.

for the others on this issue, I've trained a model with cellpose v3 and v2 and both worked correctly, so I think I've found all the bugs with training (for example it was only training with up to 8 images in the first v3 release, sorry about that!). I was also able to apply a v3 model in v2 and vice versa.

for issues that may be specific to the dataset, it could be normalization related (differences between crops and whole images), see some advice here: https://cellpose.readthedocs.io/en/latest/train.html#re-training-a-model. I would also double-check the channels are correct when training the model. And you can try decreasing the cell probability to find more cells.
